rxnetty-common/src/main/java/io/reactivex/netty/util/CollectBytes.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.reactivex.netty.util;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.CompositeByteBuf;
+import io.netty.buffer.Unpooled;
+import rx.Observable;
+import rx.Observable.Transformer;
+import rx.exceptions.OnErrorThrowable;
+import rx.functions.Action2;
+import rx.functions.Func0;
+import rx.functions.Func1;
+import rx.functions.Func2;
+
+/**
+ * An {@link Observable.Transformer} to collect a stream of {@link ByteBuf ByteBufs} into a single
+ * ByteBuf.
+ */
+public class CollectBytes implements Transformer<ByteBuf, ByteBuf> {
+
+    private static final CollectBytes INSTANCE = new CollectBytes();
+
+    /**
+     * Collect all emitted ByteBufs into a single ByteBuf.
+     */
+    public static Transformer<ByteBuf, ByteBuf> all() {
```

Comment:
Thinking about this more, I think we can cap `all()` on `Integer.MAX_VALUE`. This will have sane limits (2GB) on the buffer and reuse the code for both methods, WDYT?